packages/synapse-react-client/src/components/widgets/query-filter/CombinedRangeFacetFilter.tsx

```diff
@@ -0,0 +1,157 @@
+import { Collapse } from '@mui/material'
+import React, { useEffect } from 'react'
+import { useState } from 'react'
+import { ColumnType } from '@sage-bionetworks/synapse-types'
+import { FacetColumnResultRange } from '@sage-bionetworks/synapse-types'
+import { RadioGroup } from '../RadioGroup'
+import { Range, RangeValues } from '../Range'
+import { FacetFilterHeader } from './FacetFilterHeader'
+import { RadioValuesEnum, getRadioValue, options } from './RangeFacetFilter'
+import dayjs from 'dayjs'
+import { RangeSlider } from '../RangeSlider/RangeSlider'
+
+export type CombinedRangeFacetFilterProps = {
+  facetResults: FacetColumnResultRange[]
+  label: string
+  columnType: ColumnType
+  onChange: (range: (RadioValuesEnum | number | string | undefined)[]) => void
+  collapsed?: boolean
+}
+
+/**
+ * Inclusive range selector across two columns
+ * Written for the ELITE portal cohort builder, may have other uses.
+ * The following diagram shows how the Range Selector min and max values are used
+ * to define the facet range values:
+ * 
+
+                            Range Selector      
+                  min ├────────────────────────┤ max
+                                                
+                          Min Column facet      
+ Column Min value ◄────────────────────────────┤  Range Selector max
+                                                
+                          Max Column facet
+ Range Selector min   ├────────────────────────────► Column Max value
+                       
+
+*/
+export const CombinedRangeFacetFilter: React.FunctionComponent<
+  CombinedRangeFacetFilterProps
+> = ({
+  facetResults,
+  label,
+  columnType,
+  onChange,
+  collapsed = false,
+}: CombinedRangeFacetFilterProps) => {
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(collapsed)
+  const {
+    columnMin: col1Min,
+    // columnMax: col1Max  // not used
+    selectedMin: col1SelectedMin,
+    selectedMax: col1SelectedMax,
+  } = facetResults[0]
+  const {
+    //columnMin: col2Min,  // not used
+    columnMax: col2Max,
+    selectedMin: col2SelectedMin,
+    selectedMax: col2SelectedMax,
+  } = facetResults[1]
+
+  const isAnyValue =
+    !col1SelectedMin && !col1SelectedMax && !col2SelectedMin && !col2SelectedMax
+  const selectedMin = col2SelectedMin || col1Min
+  const selectedMax = col1SelectedMax || col2Max
+
+  const rangeType = columnType === 'DOUBLE' ? 'number' : 'date'
+
+  const handleRadioGroupChange = (
+    radioValue: RadioValuesEnum,
+    onChangeCallback: (
+      range: (RadioValuesEnum | number | string | undefined)[],
+    ) => void,
+  ) => {
+    setRadioValue(radioValue)
+
+    if (radioValue !== RadioValuesEnum.RANGE) {
+      onChangeCallback([radioValue, radioValue])
+    }
+  }
+
+  const [radioValue, setRadioValue] = useState(
+    getRadioValue(selectedMin, isAnyValue),
+  )
+
+  useEffect(() => {
+    setRadioValue(getRadioValue(selectedMin, isAnyValue))
+  }, [selectedMin, isAnyValue])
```

Comment:
Bug fix for both the RangeFacetFilter and this Combined version.  In the old implementation, after you specified a range the UI would often be initialized with the radio checkbox on "Any" (but with the correct range slider if you selected the Range radio!).  Be sure to update the radio whenever the input changes (cannot rely on the default value on mount only)